File: version_hs/src/Forms/LoginForm.hs
module Forms.LoginForm (loginForm) where

import User
import Config

import Forms.ChatForm

loginForm :: AppData -> IO MenuOption
loginForm (accountList, _) = do
  titleText " [LOGIN]        "
  putStr "Login (or :q): "
  login <- getLine

  let
    findUser = filter (\x -> login == ulogin x) accountList

  if login == ":q" then pure $ MenuNew

  else if findUser == [] then do
    pure $ MenuErr "No account"

  else do
    successText $ "Login success"

    titleText " [CHAT]         "
    chatForm $ (accountList, findUser !! 0)

File: version_py/src/login_form.py
from src.config import *
from src.chat_form import *




def login_form(account_list):
    print(term.clear + term.home)
    gui_wrapper("LOGIN", "*")
    while True:
        name_inp = get_inp("Введи имя заебал: (:q - выход) ")
        if name_inp == ":q":
            return True
        
        find_user = find_user_name(account_list, name_inp)
        if len(find_user) == 0:
            error_text("Пользователь не найден")
            continue

        chat_form(account_list, find_user[0])
        return False